src/code/z_sram_NES.c

```diff
@@ -1,67 +1,1663 @@
+        fileChooseCtx->heartPieceCount[fileChooseCtx->fileNum] =
+            (gSaveContext.save.inventory.questItems & 0xF0000000) >> 0x1C;
+    }
+
+    gSaveContext.save.time = D_801F6AF0;
+    gSaveContext.unk_3F3F = D_801F6AF2;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_sram_NES/func_80146628.s")
+#endif
+
+void func_80146AA0(FileChooseContext* fileChooseCtx2, SramContext* sramCtx) {
+    s32 phi_v0;
+    u16 i;
+    FileChooseContext* fileChooseCtx = fileChooseCtx2;
+    s16 phi_a0;
+
+    if (gSaveContext.unk_3F3F) {
+        Sram_InitNewSave();
+        if (fileChooseCtx->unk_24480 == 0) {
+            gSaveContext.save.cutscene = 0xFFF0;
+        }
+
+        for (phi_v0 = 0; phi_v0 != ARRAY_COUNT(gSaveContext.save.playerData.playerName); phi_v0++) {
+            gSaveContext.save.playerData.playerName[phi_v0] =
+                fileChooseCtx->unk_24414[fileChooseCtx->unk_24480][phi_v0];
+        }
+
+        gSaveContext.save.playerData.newf[0] = 'Z';
+        gSaveContext.save.playerData.newf[1] = 'E';
+        gSaveContext.save.playerData.newf[2] = 'L';
+        gSaveContext.save.playerData.newf[3] = 'D';
+        gSaveContext.save.playerData.newf[4] = 'A';
+        gSaveContext.save.playerData.newf[5] = '3';
+
+        gSaveContext.save.checksum = Sram_CalcChecksum(&gSaveContext.save, sizeof(Save));
+
+        Lib_MemCpy(*sramCtx->saveBuf, &gSaveContext.save, sizeof(Save));
+        Lib_MemCpy(&(*sramCtx->saveBuf)[0x2000], &gSaveContext.save, sizeof(Save));
+
+        for (i = 0; i < ARRAY_COUNT(gSaveContext.save.playerData.newf); i++) {
+            fileChooseCtx->newf[fileChooseCtx->unk_24480][i] = gSaveContext.save.playerData.newf[i];
+        }
+
+        fileChooseCtx->unk_2440C[fileChooseCtx->unk_24480] = gSaveContext.save.playerData.deaths;
+
+        for (i = 0; i < ARRAY_COUNT(gSaveContext.save.playerData.playerName); i++) {
+            fileChooseCtx->unk_24414[fileChooseCtx->unk_24480][i] = gSaveContext.save.playerData.playerName[i];
+        }
+
+        fileChooseCtx->healthCapacity[fileChooseCtx->unk_24480] = gSaveContext.save.playerData.healthCapacity;
+        fileChooseCtx->health[fileChooseCtx->unk_24480] = gSaveContext.save.playerData.health;
+        fileChooseCtx->unk_24454[fileChooseCtx->unk_24480] = gSaveContext.save.inventory.dungeonKeys[9];
+        fileChooseCtx->unk_24444[fileChooseCtx->unk_24480] = gSaveContext.save.inventory.questItems;
+        fileChooseCtx->unk_24458[fileChooseCtx->unk_24480] = gSaveContext.save.time;
+        fileChooseCtx->unk_24460[fileChooseCtx->unk_24480] = gSaveContext.save.day;
+        fileChooseCtx->unk_24468[fileChooseCtx->unk_24480] = gSaveContext.save.isOwlSave;
+        fileChooseCtx->rupees[fileChooseCtx->unk_24480] = gSaveContext.save.playerData.rupees;
+        fileChooseCtx->unk_24474[fileChooseCtx->unk_24480] =
+            (gSaveContext.save.inventory.upgrades & gUpgradeMasks[4]) >> gUpgradeShifts[4];
```

Comment:
```suggestion
        fileChooseCtx->unk_24474[fileChooseCtx->unk_24480] = CUR_UPG_VALUE(UPG_WALLET);
```

Reply:
I haven't changed this because the function is currently non matching, and I though it may be better for whoever tries to match it to have the raw statement instead of the macro.
I can change it if it is preferred tho

Reply:
Ah I didn't notice it was in a non-matching. Yeah you can leave it as it was. You could also add a comment above with ` = CUR_UPG_VALUE(UPG_WALLET);` to save a look-up later but up to you